sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/LogicalRelation.scala

```diff
@@ -33,7 +33,8 @@ import org.apache.spark.util.Utils
 case class LogicalRelation(
     relation: BaseRelation,
     expectedOutputAttributes: Option[Seq[Attribute]] = None,
-    metastoreTableIdentifier: Option[TableIdentifier] = None)
+    metastoreTableIdentifier: Option[TableIdentifier] = None,
+    inheritedStats: Option[Statistics] = None)
```

Comment:
having `metastoreTableIdentifier` inside `LogicalRelation` means we already hack the table concept into `LogicalRelation`, how about we go further and put `CatalogTable` in `LogicalRelation`? I think it's better than taking a `TableIdentifier` and a `Statistics` object.